cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/operationrunner/InMemorySourceControlOperationRunner.java

```diff
@@ -201,5 +208,48 @@ private Path validateAppConfigRelativePath(RepositoryManager repositoryManager,
     }
     return filePath;
   }
+
+  @Override
+  public ListAppsResponse list(NamespaceId namespace, RepositoryConfig repoConfig) throws ListAppsFailureException,
+    AuthenticationConfigException, NotFoundException {
+    try (RepositoryManager repositoryManager = repoManagerFactory.create(namespace, repoConfig)) {
+      String currentCommit = repositoryManager.cloneRemote();
+      Path basePath = repositoryManager.getBasePath();
+
+      if (!Files.exists(basePath)) {
+        throw new NotFoundException(String.format("Missing repository base path %s", basePath));
+      }
+
+      FileFilter configFileFilter = getConfigFileFilter();
+      List<File> configFiles = DirUtils.listFiles(basePath.toFile(), configFileFilter);
+
+      List<ListAppResponse> responses = new ArrayList<>();
+      for (File configFile : configFiles) {
+        String applicationName = FileUtils.getNameWithoutExtension(configFile.getName());
+        String fileHash = repositoryManager.getFileHash(basePath.relativize(configFile.toPath()), currentCommit);
```

Comment:
Shouldn't we handle symlinks by calling `validateAppConfigRelativePath` here?

Reply:
Good point. I was depending on the old getfilehash behaviour. Better to handle this at filefilters